### ISSUE TITLE
ERROR: Can't open dynamic library: addons/luaAPI/bin/libluaapi.linux.template_debug.x86_64.so. Error: /home/r3ddy/Scaricati/../lib/libluaapi.linux.template_debug.x86_64.so: cannot open shared object file: No such file or directory.    at: open_dynamic_library (drivers/unix/os_unix.cpp:660)

### ISSUE THREAD

**Issue by Superredstone** [face-hh/griddycode#3]
My console is getting filled with this errors, i guess that there is some kind of library that i am missing? Maybe you should add them to the requirements? 

OS: Arch Linux x86_64
Kernel: Linux 6.8.1-arch1-1
DE: KDE Plasma

```
ERROR: Can't open dynamic library: addons/luaAPI/bin/libluaapi.linux.template_debug.x86_64.so. Error: /home/r3ddy/Scaricati/../lib/libluaapi.linux.template_debug.x86_64.so: cannot open shared object file: No such file or directory.
   at: open_dynamic_library (drivers/unix/os_unix.cpp:660)
ERROR: GDExtension dynamic library not found: addons/luaAPI/bin/libluaapi.linux.template_debug.x86_64.so
   at: open_library (core/extension/gdextension.cpp:719)
ERROR: Failed loading resource: res://addons/luaAPI/luaAPI.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:274)
ERROR: Error loading extension: res://addons/luaAPI/luaAPI.gdextension
   at: load_extensions (core/extension/gdextension_manager.cpp:234)
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
Vulkan API 1.3.277 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3050 Laptop GPU

SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:340)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:341)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:353)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:354)
ERROR: Failed to load script "res://Scripts/lua_singleton.gd" with error "Parse error".
   at: load (modules/gdscript/gdscript.cpp:2788)
ERROR: Failed to instantiate an autoload, script 'res://Scripts/lua_singleton.gd' does not inherit from 'Node'.
   at: start (main/main.cpp:3207)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:340)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:341)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:353)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:354)
SCRIPT ERROR: Compile Error:
          at: GDScript::reload (res://Scripts/file_manager.gd:-1)
ERROR: Failed to load script "res://Scripts/file_manager.gd" with error "Compilation failed".
   at: load (modules/gdscript/gdscript.cpp:2788)
SCRIPT ERROR: Parse Error: Could not resolve member "lua".
          at: GDScript::reload (res://Scripts/settings.gd:71)
ERROR: Failed to load script "res://Scripts/settings.gd" with error "Parse error".
   at: load (modules/gdscript/gdscript.cpp:2788)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:340)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:341)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:353)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:354)
SCRIPT ERROR: Compile Error:
          at: GDScript::reload (res://Scripts/file_dialog.gd:-1)
ERROR: Failed to load script "res://Scripts/file_dialog.gd" with error "Compilation failed".
   at: load (modules/gdscript/gdscript.cpp:2788)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:281)
SCRIPT ERROR: Parse Error: Could not find type "LuaAPI" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Identifier "LuaAPI" not declared in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:282)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:340)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:341)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:353)
SCRIPT ERROR: Parse Error: Could not find type "LuaError" in the current scope.
          at: GDScript::reload (res://Scripts/lua_singleton.gd:354)

```

**Comment by face-hh** [comment]
I assume it was because I shipped the executable without the .so library, could you try the updated release for Linux?

**Comment by Superredstone** [comment]
Ok, the error has been fixed but now a new error appeared and nothing is displayed
```
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
Vulkan API 1.3.277 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3050 Laptop GPU

/
ERROR 6: error while opening file 'user://langs/.lua'
ERROR 6: error while opening file 'user://themes/One Dark Pro Darker.lua'

```
![immagine](https://github.com/face-hh/griddycode/assets/56410215/4a4d82e4-9af5-48e5-b460-2e5967998f95)
![2024-03-19_17-35](https://github.com/face-hh/griddycode/assets/56410215/20651b5a-b14e-4b8a-8137-c6d70e9d9091)

**Comment by face-hh** [comment]
Fixed in v1.0.3
